### PR TITLE
Make era-test-node a library

### PR DESCRIPTION
**Pull request by mm-zk**
# What :computer: 
* Adding lib.rs that allows other rust programs to include era-test-node as a library
* moved ShowCalls into this crate (before it was in main.rs)

# Why :hand:
* This allows other rust programs (for example forge testing) to include era test node as a library.

# Evidence :camera:
You can include era-test-node from other crates.
```
[dependencies]
era_test_node = {path = "../era-test-node"}
```

### CODE REVIEW

File: src/formatter.rs [hunk 1]
```diff
@@ -13,6 +14,34 @@ use zksync_types::{vm_trace::Call, StorageLogQuery, StorageLogQueryType, VmEvent
...
+impl FromStr for ShowCalls {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_ref() {
+            "none" => Ok(ShowCalls::None),
+            "user" => Ok(ShowCalls::User),
+            "system" => Ok(ShowCalls::System),
+            "all" => Ok(ShowCalls::All),
+            _ => Err(()),
```
Comment (idea404) [inline src/formatter.rs:34]: A little of a vague error, would be better to add some description

File: src/node.rs [hunk 2]
```diff
@@ -98,6 +102,58 @@ pub struct TxExecutionInfo {
...
+impl FromStr for ShowStorageLogs {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_ref() {
+            "none" => Ok(ShowStorageLogs::None),
+            "read" => Ok(ShowStorageLogs::Read),
+            "write" => Ok(ShowStorageLogs::Write),
+            "all" => Ok(ShowStorageLogs::All),
+            _ => Err(()),
```
Comment (idea404) [inline src/node.rs:122]: Same as with `ShowCalls`

File: src/node.rs [hunk 3]
```diff
@@ -98,6 +102,58 @@ pub struct TxExecutionInfo {
...
+}
+
+impl FromStr for ShowVMDetails {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_ref() {
+            "none" => Ok(ShowVMDetails::None),
+            "all" => Ok(ShowVMDetails::All),
+            _ => Err(()),
```
Comment (idea404) [inline src/node.rs:146]: same as with `ShowCalls`

**Review by idea404 — COMMENTED** [review]
Minor suggestions. It looks to me as if the new enums help fulfil a formatting purpose, in which case it might make sense to place them in `formatter.rs`

File: src/node.rs [hunk 4]
```diff
@@ -98,6 +102,58 @@ pub struct TxExecutionInfo {
     pub result: VmTxExecutionResult,
 }
 
+#[derive(Debug, Parser, Clone, clap::ValueEnum, PartialEq, Eq)]
```
[the remark this answers is not in the document]
Comment (mm-zk) [reply src/node.rs:136]: good point. Moved.

File: src/formatter.rs (hunk 1 above)
Comment (mm-zk) [reply src/formatter.rs]: good point. Done

**Review by MexicanAce — APPROVED** [review]
LGTM

**Pull request merged**